### PR TITLE
Fixes for DWRF encryption

### PR DESCRIPTION
**Pull request by rschlussel**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java [hunk 1]
```diff
@@ -623,19 +629,19 @@ public void close()
...
-                        .stream()
-                        .map(DwrfMetadataWriter::toColumnStatistics)
-                        .map(AbstractMessageLite::toByteArray)
-                        .collect(toImmutableList());
-
                 DwrfProto.FileStatistics fileStatistics = toFileStatistics(encryptedStats.get(i));
-                byte[] serializedStats = fileStatistics.toByteArray();
+                DwrfDataEncryptor dwrfDataEncryptor = dwrfEncryptionInfo.getEncryptorByGroupId(i);
+                OrcOutputBuffer buffer = new OrcOutputBuffer(compression, Optional.of(dwrfDataEncryptor), maxCompressionBufferSize);
+                fileStatistics.writeTo(buffer);
```
Comment (zzhao0) [inline presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java:635]: do we have unit tests for stats of nested types?

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java [hunk 2]
```diff
@@ -461,13 +456,21 @@ private void writeChunkToOutputStream(byte[] chunk, int offset, int length)
 
         int compressedSize = compressor.compress(chunk, offset, length, compressionBuffer, 0, compressionBuffer.length);
         if (compressedSize < length) {
+            if (dwrfEncryptor.isPresent()) {
```
Comment (zzhao0) [inline presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java:459]: It's unclear to me how you handle encrypted but uncompressed file. Would be good to throw in that case if not supported at the moment

File: presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java [hunk 3]
```diff
@@ -462,11 +467,19 @@ private void advance()
         if (isUncompressed) {
             buffer = ensureCapacity(buffer, chunkLength);
             length = compressedSliceInput.read(buffer, 0, chunkLength);
+            if (dwrfDecryptor.isPresent()) {
```
Comment (zzhao0) [inline presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java:470]: This logic is correct for encrypted and compressed file (and a single chunk/page is uncompressed). We need to handle encrypted but uncompressed file.

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java (hunk 1 above)
Comment (rschlussel) [reply presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java:635]: We don't have any fileStatistics tests (Presto doesn't use filestatistics for dwrf, so none of our tests are testing the functionality). Let me think about how to write some kind of test though and I'll add it as a follow on

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java (hunk 2 above)
Comment (rschlussel) [reply presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java:459]: As per offline conversation I will add a follow up to support this.  I previously thought that wasn't supported by the file format.

**Pull request merged**